warehouse/query-core/src/main/java/datawave/query/jexl/DatawavePartialInterpreter.java

```diff
@@ -0,0 +1,887 @@
+        super(jexl, aContext, strictFlag, silentFlag);
+        this.incompleteFields = incompleteFields;
+        this.callback = callback;
+        this.callback.reset();
+    }
+    
+    /**
+     * Sets the callback state if an incomplete field contributed to a positive evaluation
+     *
+     * @param node
+     *            the JexlNode
+     * @return true if the query matched the document
+     */
+    @Override
+    public Object interpret(JexlNode node) {
+        Object matched = super.interpret(node);
+        if (matched instanceof State) {
+            State state = (State) matched;
+            if (state.matched && state.fromUnknown) {
+                callback.setUsed(true);
+            }
+        } else {
+            throw new IllegalStateException("result of interpret was not STATE, it was " + matched.getClass().getSimpleName());
+        }
+        return matched;
+    }
+    
+    /**
+     * Returns true if the provided JexlNode contains an identifier that is also present in the set of incomplete fields
+     *
+     * @param node
+     *            the JexlNode
+     * @return true if incomplete fieldname found, false otherwise
+     */
+    private boolean isIncomplete(JexlNode node) {
+        for (String field : JexlASTHelper.getIdentifierNames(node)) {
+            field = JexlASTHelper.deconstructIdentifier(field);
+            // if the field is not in the context, we cannot evaluate it.
+            return incompleteFields.contains(field) && context.has(field);
+        }
+        return false;
+    }
+    
+    /**
+     * Determines if the result of a script execution evaluates to true
+     *
+     * @param scriptExecuteResult
+     *            the result of visiting a query node
+     * @return true if the object matched
+     */
+    public static boolean isMatched(Object scriptExecuteResult) {
+        if (scriptExecuteResult instanceof State) {
+            State state = (State) scriptExecuteResult;
+            return state.matched;
+        } else {
+            return DatawaveInterpreter.isMatched(scriptExecuteResult);
+        }
+    }
+    
+    /**
+     *
+     * @param node
+     *            an ASTFunctionNode
+     * @param data
+     *            some data
+     * @return a result
+     */
+    @Override
+    public Object visit(ASTFunctionNode node, Object data) {
+        
+        // get the namespace
+        FunctionJexlNodeVisitor visitor = new FunctionJexlNodeVisitor();
+        visitor.visit(node, null);
+        
+        // content functions should always have the field as the first argument
+        if (visitor.namespace().equals(ContentFunctions.CONTENT_FUNCTION_NAMESPACE)) {
+            JexlNode first = visitor.args().get(0);
+            if (first instanceof ASTIdentifier) {
+                String field = JexlASTHelper.deconstructIdentifier((ASTIdentifier) first);
+                if (context.has(field) && incompleteFields.contains(field)) {
```

Comment:
does it matter if the context has the field? Is it enough that the incomplete fields contains it?

Reply:
Yes, the field's presence in the context matters. See comment above.